taskvine/src/tools/vine_plot_txn_log

```diff
@@ -687,31 +713,55 @@ class TxnPlotWorkers(TxnPlot):
         self.bh(fig, ft["slot"], ft["RETRIEVED"] - 0.5, 0.5, self.legend["tasks failures"])
 
         return worker["slot"].max() + 1
-
+    
+    
+    def plot_by_worker_dask(self, fig, worker_tasks, worker_times, worker_xfers, base_slot):
+        worker_tasks["slot"] += (base_slot + 0.2)
+        retrieved_tasks = worker_tasks[~worker_tasks["RETRIEVED"].isna()]
+        cache_updates = worker_xfers[ worker_xfers["direction"] == "CACHE_UPDATE" ]
+        slot_height = 1
+
+        # plot the red bar: cache updates
+        self.bh(fig, base_slot, cache_updates["start_time"], cache_updates["wall_time"], 
+                self.legend_dask["data transfers"], height=slot_height, type="cu_dask")
+
+        # every following task's bar slot add 1
+        worker_tasks["slot"] += 1
+
+        # plot the green bar: tasks executing
+        self.bh(fig, retrieved_tasks["slot"], retrieved_tasks["time_worker_start"], 
+                retrieved_tasks["time_worker_end"] - retrieved_tasks["time_worker_start"], 
+                self.legend_dask["tasks executing"], height=slot_height, type="te_dask")
+        
+        # determine the following starting slot for tasks in the next worker
+        return worker_tasks["slot"].max() + 0.2
 
     def plot(self, fig, manager):
-        ts = manager.tasks
-        xs = manager.transfers
-        ws = manager.workers
-
-        gs = ts.groupby("worker_id", sort=False)
-        by_min_connects = ws.groupby("worker_id")["CONNECTION"].min().sort_values().index
-        gsd = {wid: g  for (wid, g) in gs}
-
-        base_slot=0
-        for wid in by_min_connects:
-            worker = gsd.get(wid, None)
-            worker_times = ws[ws["worker_id"] == wid]
-            worker_xfers = xs[xs["worker_id"] == wid]
-
-            if worker is not None:
-                base_slot = self.plot_by_worker(fig, worker, worker_times, worker_xfers, base_slot)
+        tasks = manager.tasks
```

Comment:
Making the names clearer is fine, however do not include these unrelated changes in this pr. Otherwise it makes it harder to evaluate the main objective of the pr, and may introduce unrelated bugs.

Reply:
Sure! And could you give me a hint about how to remove those unrelated commits?